articles/most-useful-nodejs-packages/index.md

```diff
@@ -0,0 +1,366 @@
+
+#### 7. JSHint
+[JSHint](https://jshint.com/about/) is a static analysis tool for Javascript. It detects errors and potential problems in code. Linting tools help Node.js developers to analyze common bugs in their code without their focused attention. Linters analyses the code for them.
+
+JSHint scans your Node.js program and reports commonly made mistakes and potential bugs such as syntax errors, leaking variables, implicit type, and conventions etc.
+
+[`ESLint`](https://www.section.io/engineering-education/node-eslint/) is similar to JShint package, used for linting (enhancing code quality).
+
+### 8. Morgan
+[Morgan](https://www.npmjs.com/package/morgan) is an HTTP request logger middleware for Node.js applications, named after [Dexter](https://en.wikipedia.org/wiki/Dexter_Morgan). Morgan gives you insights on how your app is being used and alerts you on potential errors and issues that could be threats to your application. Morgan is considered the most reliable HTTP logger by Node.js developers.
+
+You can choose to use [`Winston`](https://www.npmjs.com/package/winston) as an alternative library for logging.
+
+### 9. Karma
+As a Node.js developer, you need to test your application to make sure it is stable and reliable with good performance. [Karma](https://www.npmjs.com/package/karma) is the tool for this job. The main aim of karma is to provide developers a productive testing environment.
+
+#### Why You Should Choose Karma
+-   Tests code in real browsers.
+-   Tests code in multiple browsers (desktop, mobile phones and tablets).
+-   It tests on real devices such as mobile phones, tablets or even headless PhantomJS.
+-   Controls the whole testing workflow from your command or IDE.
+-   Karma executes your tests after every save.
+-   It executes tests locally during development.
+-   It executes your tests on a continuous integration server. [Refer to  Karma](https://www.npmjs.com/package/karma).
+
+#### Other Node.js Testing Libraries Include
+-   [`Mocha`](https://www.npmjs.com/package/mocha) - helps you to carry out asynchronous tests.
+-   [`Chai`](https://www.chaijs.com/) - is a BDD/TDD assertion library for Node.js and browsers that can be paired with any JavaScript testing framework.
+-   [`Jest`](https://jestjs.io/) - JavaScript testing framework which focuses on simplicity.
+
+### 10. MySQL
+[MySQL](https://en.wikipedia.org/wiki/MySQL) is a Node.js client for the MySQL protocol. Before using MySQL to connect to your database, ensure you have [MySQL installed and configured](https://dev.mysql.com/doc/mysql-getting-started/en/) in your machine. Then create a database and database table that you can work with. [Check instructions on how to create MySQL database and tables](https://www.guru99.com/how-to-create-a-database.html).
+
+**An Example on How to Use MySQL with Node.js**
+```js
+//use require() to include MySql library
+var mysqldb = require('mysql');
```

Comment:
Trying to run the code after installing MySQL and received this error: `Error: ER_NOT_SUPPORTED_AUTH_MODE: Client does not support authentication protocol requested by server; consider upgrading MySQL client`

Reply:
@louisefindlay23 Found out that if the latest MySQL server installed you might end up getting that error.

Reply:
The updated code block works

Reply:
@kimkimani, running the updated code block results in `ReferenceError: con is not defined`

Reply:
@louisefindlay23 
should work now
var con = mysqldb.createConnection({

Reply:
I'm afraid I still get, `Error: ER_NOT_SUPPORTED_AUTH_MODE: Client does not support authentication protoc
ol requested by server; consider upgrading MySQL client` @kimkimani.

Reply:
Okay @louisefindlay23 
I may consider removing the code block then, though it is working on my side, I can't figure out why.
since am planning to do a Node MySQL article I will consider digging deeper on this.

Reply:
I'll see if someone else can reproduce it in case it's isolated to my installation.

Reply:
updated @louisefindlay23

Reply:
@kimkimani, I've asked another engineer to look at the code block to see if they can reproduce the error or if the issue is isolated to my installation but I've noticed you've removed the MySQL code.

Reply:
Yeah, I went the thump up on the above comment before your comment got updated.

Reply:
@louisefindlay23 I've done a revert

Reply:
@nadivgold - could you try to replicate this error that Louise is seeing (when you get a moment) - we just want to make sure the code block works like intended - before we move forward. 
Thank you 🚀